Eatery/Views/Eateries/MealItemTableViewCell.swift

```diff
@@ -2,15 +2,23 @@
 //  MealItemTableViewCell.swift
 //  Eatery
 //
-//  Created by William Ma on 10/24/18.
-//  Copyright © 2018 CUAppDev. All rights reserved.
+//  Created by Ethan Fine on 2/9/20.
+//  Copyright © 2020 CUAppDev. All rights reserved.
 //
 
 import UIKit
+import UserNotifications
 
 class MealItemTableViewCell: UITableViewCell {
 
-    let nameLabel = UILabel()
+    private var favoriteButton: UIButton?
+    private(set) var menuItem: Menu.Item?
+    private let nameLabel = UILabel()
+    
+    private var favoriteButtonAttributes: (imageName: String, tintColor: UIColor)? {
```

Comment:
Can you try making this into a non-optional?

Reply:
There can be a MealItemTableViewCell that does not correspond to a particular menu item, for example the no item cell. I believe it's best to keep it optional.

Reply:
It seems this is only used when we have a meal item. Maybe instead of having it be a property, can we have this be a function that takes in a non-optional menu item?

Reply:
This is in fact used as well with no item. Notice in line 142-148 of CampusEateryMealTableViewController, a MealItemTableViewCell is used to display that there is no item. This must be maintained as a property because it is then later used by the tappedFavorite function to toggle the favorability of menuItem.

Reply:
What I'm asking is: why can't this be a function a la:
```
func favoriteButtonAttributes(of menuItem: Menu.Item) -> (imageName: String, tintColor: UIColor)? {
...
}
```
Or, since the favorite property is the only one that's used:
```
```

Reply:
Sure, this could be implemented as a function, however, to what advantage? Would still need to hold an optional reference to menuItem. Also, the button attributes are more of a property of the cell than something that depends on an arbitrary isFavorite or menuItem to called into a function call (only menuItem or menuItem.isFavorite would ever be passed in), so I still feel it should be a computed property.

Reply:
We could remove a whole bunch of force unwraps is the main benefit.

Reply:
Ok, this sounds good as whenever favoriteButtonAttributes is called it won't have to be force unwrapped, since menuItem will have already been unwrapped at this point.